kong/templates/nginx_kong.lua

```diff
@@ -56,6 +56,9 @@ lua_shared_dict kong_rate_limiting_counters 12m;
 > if database == "cassandra" then
 lua_shared_dict kong_cassandra      5m;
 > end
+> if role == "admin" then
+lua_shared_dict kong_clustering     5m;
+> end
```

Comment:
If the role is proxy, then should other shms related to DBs be disabled? Like cassandra, db_cache_miss etc?

Reply:
It seems that the `cassandra` shdict is disabled with proxy mode already, since it uses `database = "off"`:

```
> if database == "cassandra" then
lua_shared_dict kong_cassandra      5m;
> end
```

Reply:
What about cache_miss shms though? I don't think they are used in DB-less mode.

Reply:
That is a DB-less specific issue, let's fix it elsewhere in an interest to not overly complicate this PR.

Reply:
@hbagdi cache_miss shms are used in DB-less as well. Anything that uses `kong.cache:get` calls (core, plugins) will use both the positive and negative caches as normal.